src/hb-set.h

```diff
@@ -60,126 +60,147 @@ typedef struct hb_set_t hb_set_t;
 HB_EXTERN hb_set_t *
 hb_set_create (void);
 
+/**
+ * Creates a set initially populated with num_codepoint codepoints, which must
+ * be in increasing order.
+ *
+ * Since: REPLACEME
+ */
+HB_EXTERN hb_set_t *
+hb_set_create_from_sorted (const hb_codepoint_t  *sorted_codepoints,
+                           unsigned int          num_codepoints);
+
+/**
+ * Exports the contents of the set to an array of hb_codepoint_t.
+ * codepoints[0] .. codepoints[hb_set_get_population(set) - 1] will be
+ * written to; you must ensure the size of the memory codepoints points to.
+ * The results will be in ascending order.
+ *
+ * Since: REPLACEME
+ */
+HB_EXTERN void
+hb_set_export_array (const hb_set_t  *set,
+                     hb_codepoint_t  *codepoints);
+
 HB_EXTERN hb_set_t *
 hb_set_get_empty (void);
 
 HB_EXTERN hb_set_t *
-hb_set_reference (hb_set_t *set);
+hb_set_reference (hb_set_t  *set);
 
 HB_EXTERN void
-hb_set_destroy (hb_set_t *set);
+hb_set_destroy (hb_set_t  *set);
 
 HB_EXTERN hb_bool_t
-hb_set_set_user_data (hb_set_t           *set,
-		      hb_user_data_key_t *key,
-		      void *              data,
-		      hb_destroy_func_t   destroy,
-		      hb_bool_t           replace);
+hb_set_set_user_data (hb_set_t            *set,
+                      hb_user_data_key_t  *key,
```

Comment:
Can you undo all the formatting changes on unrelated code. Notably the default style for harfbuzz is to use mixed tabs and spaces. It looks like things have been converted to just spaces.

Reply:
Done